### PR TITLE
[MNG-7468] Check unsupported plugins parameters in configuration

### PR DESCRIPTION
**Pull request by slawekjaranowski**

### CODE REVIEW

File: core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/TouchMojo.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            {
+                dir.mkdirs();
+            }
+
+            File touch = new File( dir, file );
+
+            try ( OutputStreamWriter w = new OutputStreamWriter(
+                new FileOutputStream( touch, true ), StandardCharsets.UTF_8 ) )
+            {
+                w.write( file );
```
Comment (michael-o) [inline core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/TouchMojo.java:73]: A `touch` produces a zero byte file, this one don't.

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7468UnsupportedPluginsParametersTest.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        Verifier verifier = newVerifier( new File( testDir, project ).getAbsolutePath() );
+        verifier.executeGoal( "validate" );
+        verifier.verifyErrorFreeLog();
+        verifier.resetStreams();
+
+        List<String> logLines = verifier.loadFile( verifier.getBasedir(), verifier.getLogFileName(), false );
+        return findUnknownWarning( logLines );
+    }
+
+    private void assertWarningPresents( List<String> warnLines )
```
Comment (michael-o) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7468UnsupportedPluginsParametersTest.java:158]: Rather `assertWarningContains`, no?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7468UnsupportedPluginsParametersTest.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+        assertTrue( warnLines.remove(
+            "[WARNING] Parameter 'invalidXml' is unknown for plugin 'maven-it-plugin-configuration:2.1-SNAPSHOT:touch (default)'" ) );
+
+        assertTrue( "Not verified line: " + warnLines, warnLines.isEmpty() );
+    }
+
+    private List<String> findUnknownWarning( List<String> logLines )
+    {
+        Pattern pattern = Pattern.compile( "\\[WARNING] Parameter .* is unknown.*" );
```
Comment (michael-o) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7468UnsupportedPluginsParametersTest.java:171]: WE rather require content: `.+`, no?

File: core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/TouchMojo.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    {
+        try
+        {
+            if ( !dir.exists() )
+            {
+                dir.mkdirs();
+            }
+
+            File touch = new File( dir, file );
+            touch.createNewFile();
```
Comment (michael-o) [inline core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/TouchMojo.java:66]: https://www.youtube.com/watch?v=_NNYI8VbFyY

File: core-it-suite/src/test/resources/mng-7468-unsupported-params/config-build-execution/pom.xml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+          <execution>
+            <goals>
+              <goal>touch</goal>
+            </goals>
+            <configuration>
+              <invalidValue>value</invalidValue>
+              <invalidXml>
+                <item>value</item>
+              </invalidXml>
+            </configuration>
```
Comment (michael-o) [inline core-it-suite/src/test/resources/mng-7468-unsupported-params/config-build-execution/pom.xml:50]: This confuses me a bit. We don't validate the value, but only the param, no? So why not `invalidParam`? What am I missing?
The next block is an Xpp3Dom, so I guess `invalidXml` is correct.

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7468UnsupportedPluginsParametersTest.java (hunk 3 above)
Comment (slawekjaranowski) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7468UnsupportedPluginsParametersTest.java:171]: in this case will be the same ... we searching lines of this pattern and in next step specific line values are compared

Comment (michael-o) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7468UnsupportedPluginsParametersTest.java:171]: OK, I see.

File: core-it-suite/src/test/resources/mng-7468-unsupported-params/config-build-execution/pom.xml (hunk 5 above)
Comment (slawekjaranowski) [reply core-it-suite/src/test/resources/mng-7468-unsupported-params/config-build-execution/pom.xml:50]: ok, `invalidParam` will be more precise - yes we check parameter names not values

`invalidXml` can be configuration for complex custom object or collections like list

I don't now  - can plugin have access to raw `Xpp3Dom` if not defined parameter in plugin.xml descriptor

File: core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/TouchMojo.java (hunk 4 above)
Comment (michael-o) [reply core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/TouchMojo.java:66]: No, just wanted to add some fun. :-D

File: core-it-suite/src/test/resources/mng-7468-unsupported-params/config-build-execution/pom.xml (hunk 5 above)
Comment (slawekjaranowski) [reply core-it-suite/src/test/resources/mng-7468-unsupported-params/config-build-execution/pom.xml:50]: both new?
 - `invalidValue` -> `invalidParam`
 - ?

Comment (slawekjaranowski) [reply core-it-suite/src/test/resources/mng-7468-unsupported-params/config-build-execution/pom.xml:50]: done

**Pull request merged**